Some suggested updates to the major release blog post (#35)
jupyter-book/blog: posts/2025-11-04-why-make-a-major-release.md

---
title: "Why we made a major release for Jupyter Book 2 instead of creating a new package"
date: 2025-11-04
license: CC-BY-4.0
authors:
  - Jupyter Book Team
---

Last night, we [released Jupyter Book 2](https://github.com/jupyter-book/jupyter-book/releases/tag/v2.0.0).
This is a major release that introduces the [MyST Document Engine](https://mystmd.org) as the back-end infrastructure that powers Jupyter Book.
The alpha and beta versions of this release have been in progress since November 2024, and it represents a huge amount of work from the team and community that we are incredibly excited about.

> It also means that many Jupyter Books out there are now broken! 

Reviewer: ```suggestion
Unfortunately, it also means that many Jupyter Books out there are now broken.
```

In particular, users who haven't pinned their Jupyter Book versions to the `1.x` series will likely have an unexpected surprise.

> Users can make sure their version of Jupyter Book is compatible with the 1.x series with using commands like:

Reviewer: ```suggestion
Users can make sure their version of Jupyter Book is compatible with the 1.x series by using commands like:
```

```shell
$ pip install "jupyter-book<2"
OR
$ pip install "jupyter-book~=1"
```

However we know that many people do not pin versions in practice.

## Why we didn't create an entirely new package

We considered creating a new Python package to reflect this big switch - e.g., `pip install jupyter-book2`.
This has been done before in the Python system, and is a way to strictly separate the two projects.

However, having two codebases -- that are very different to each other -- can introduce a lot of confusion for users.
Importantly, it also introduces an challenging maintenance burden for developers.

We don't technically have to maintain both codebases: we could tell anyone coming to ask for help with Jupyter Book 1 that we aren't going to answer their questions.
We like helping people though!
Its tough to have to say no again and again over time.

> Additionally, having two user-facing products both called "jupyter-book" creates some confusion: community members and users can end up talking across each other without realising that there are two "jupyter book"-type things existing in parallel.

Reviewer: ```suggestion
Finally, we think that having two user-facing products both called "Jupyter Book" creates some confusion: community members and users can end up talking across each other without realising that there are two engines that exist in parallel.
```

> We didn't all agree across the core team.

Reviewer: ```suggestion
As it goes in open source projects, we discussed the various transition pathways, and acknowledge that not all team members felt equally comfortable with the chosen path—being concerned about the burden the transition imposed. 

However, we came to a consensus around taking that path, are all very enthusiastic about the new engine, and dedicated to helping our users adopt it successfully. 
```

We _all_ dislike the feeling that many of you will experience if your Jupyter Book 1 builds start to break.
We're really proud of our MyST document engine and we want you to feel excited to adopt this new version of Juptyer Book.
In our team conversations we recognised that looking at your CI fail and seeing all those :x:s will not fill you with joy.

The counter to users having a highly stressful single _day_ is our maintainers having a slightly stressful _year_.
We're all humans across this open source community.
And we're imperfect, and trying to use our time most wisely.

<!-- Can someone help me make this sentence emphasised? A pull out? or an admonition? -->
Ultimately, the team decided to go with making this major release (`pip install jupyter-book`) instead of creating a new package (`pip install jupyter-book2`).

## What we're doing to help

We know that this will result in hidden breakages in CI/CD and unexpected changes to user workflows.
We are sorry about this!

> We know that it's a huge pain to have to update your infrastructure and debug unexpected changes.

Reviewer: ```suggestion
We apologize in advance!
```

We hope that these reduce the burden on users in upgrading their books if they wish!

(And remember, you can always [downgrade to Jupyter Book 1](#downgrade-jb1) and keep using that.)

### Ask for support and guidance in our community channels

Finally, we've set up a number of [community channels](jb:community) where users and project contributors can discuss with one another and ask questions.

Here are a few dedicated spaces:

- [Our chat room](https://discord.mystmd.org/) is for quick conversations and synchronous questions.
- [Our discussions forum](https://github.com/orgs/jupyter-book/discussions) is for all kinds of general questions.
- The [Upgrading Jupyter Book category](https://github.com/orgs/jupyter-book/discussions/categories/upgrading-jupyterbook) is a special section of the forum for asking questions about upgrades.

The Jupyter Book team will be monitoring these channels for questions from folks as they hit papercuts and snags while they upgrade.

## Thanks for your support

Thanks again for using Jupyter Book and being part of our community.
We know that disruptions to your workflow are really annoying! 

> We're sorry if this decision has made your day more difficult. 

Reviewer: ```suggestion
We know that disruptions to your workflow are really annoying,
but we hope you will also find that the benefit of a brand new engine,
built on modern technology, will outweigh the disruption. 
```

> We are committed to helping, guiding, and learning with our community to make the transition as smooth as possible.

Reviewer: ```suggestion
```

---
title: "Why we made a major release for Jupyter Book 2 instead of creating a new package"
date: 2025-11-04
license: CC-BY-4.0
authors:
  - Jupyter Book Team
---

Last night, we [released Jupyter Book 2](https://github.com/jupyter-book/jupyter-book/releases/tag/v2.0.0).
This is a major release that introduces the [MyST Document Engine](https://mystmd.org) as the back-end infrastructure that powers Jupyter Book.
The alpha and beta versions of this release have been in progress since November 2024, and it represents a huge amount of work from the team and community that we are incredibly excited about.

Unfortunately, it also means that many Jupyter Books out there are now broken.
In particular, users who haven't pinned their Jupyter Book versions to the `1.x` series will likely have an unexpected surprise.

> Users can make sure their version of Jupyter Book is compatible with the 1.x series by using commands like:

Reviewer: ```suggestion
If you are not ready to try out the new version, you can reinstall a version of Jupyter Book compatible with the 1.x series by using commands like:
```

```shell
$ pip install "jupyter-book<2"
OR
$ pip install "jupyter-book~=1"
```

## Why we didn't create an entirely new package

> We considered creating a new Python package to reflect this big switch - e.g., `pip install jupyter-book2`.

Reviewer: ```suggestion
We considered creating a new Python package to reflect the big switch to a new documentation engine - e.g., `pip install jupyter-book2`.
```

This has been done before in the Python system, and is a way to strictly separate the two projects.

However, we see the engine upgrade as a continuation of the project, and not a complete replacement that warrants a new package name. We believe that it provides features that our users will appreciate, and want to make it possible for them to cross the gap between the two versions without much effort.

We do not plan on maintaining the old Sphinx-based engine, and think that releasing the new engine under the same package name signals that intent most clearly.

Finally, we think that having two user-facing products both called "Jupyter Book" creates some confusion: community members and users can end up talking across each other without realising that there are two engines that exist in parallel.

> As it goes in open source projects, we discussed the various transition pathways, in which some team members expressed reservations with the chosen path—being concerned about the burden the transition imposed. 

Reviewer 2: side-bar: I love the fact that we're explicitly saying that some folks disagree with this decision. IMO that is healthy (and appreciated from my perspective)
